src/components/MiniBlog/index.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useEffect, useState } from "react";
+import useStyles from "./styles";
+import { useTranslation } from "react-i18next";
+import {
+  Button,
+  Card,
+  CardContent,
+  CardMedia,
+  Container,
+  createStyles,
+  Grid,
+  Paper,
+  Tab,
+  Tabs,
+  Theme,
+  Typography,
+  withStyles,
+} from "@material-ui/core";
+import ReactMarkdown from "react-markdown";
+import { BLOG_KEYWORDS, VIEW_PORT } from "../../constants";
+import blogs from "../../posts.json";
+import CustomTag from "../CustomTag";
+import Slider from "react-slick";
+
+interface StyledTabProps {
+  label: string;
+  value: string;
+}
+
+interface TabProps {
+  id: string;
+  title: string;
+  blog: string;
+  description: string;
+  image: string;
+  tags: string;
+  author: string;
+  length: number;
+}
+
+const MiniBlog: React.FC = () => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const [jsonMdData, setJsonMdData] = useState<any>();
+  const [value, setValue] = React.useState("all");
+  const mobileBreakpoint = VIEW_PORT.MOBILE_BREAKPOINT;
+
+  const SampleNextArrow = (props: any) => {
+    const { className, style, onClick } = props;
+    return (
+      <div
+        className={className}
+        style={{ ...style, display: "block" }}
+        onClick={onClick}
+      >
+        <img
+          src="../Images/svg/right_arrow.svg"
+          alt={t("home.adaptorsTestimonials.nextArrowAlt")}
+        />
+      </div>
+    );
+  };
+
+  const SamplePrevArrow = (props: any) => {
+    const { className, style, onClick } = props;
+    return (
+      <div
+        className={className}
+        style={{ ...style, display: "block" }}
+        onClick={onClick}
+      >
+        <img
+          src="../Images/svg/left_arrow.svg"
+          alt={t("home.adaptorsTestimonials.previousArrowAlt")}
+        />
+      </div>
+    );
+  };
+
+  const StyledTab = withStyles((theme: Theme) =>
+    createStyles({
+      root: {
+        textTransform: "none",
+        color: theme.palette.primary.main,
+        fontWeight: theme.typography.fontWeightBold,
+        fontSize: "16px",
+        opacity: 1,
+        marginRight: theme.spacing(1),
+        "&:focus": {
+          opacity: 1,
+          color: theme.palette.warning.dark,
+        },
+        "&:active": {
+          opacity: 1,
+          color: theme.palette.warning.dark,
+        },
+      },
+    })
+  )((props: StyledTabProps) => <Tab disableRipple {...props} />);
+
+  useEffect(() => {
+    function fetchBlogs() {
```

Comment:
Can we take this function outside, convert it to a typescript method and just call the method from inside useEffect().

Reply:
updated it as per the comments.